### DISCUSSION TITLE
Полное руководство по Node.js — Изучите Node для начинающих ⚡️ Node.js с примерами кода

### DISCUSSION THREAD

**Discussion by giscus** [xsltdev/nodejsdev.ru#48] [Announcements]
# Полное руководство по Node.js — Изучите Node для начинающих ⚡️ Node.js с примерами кода

Справочник Node.js с примерами кода

https://nodejsdev.ru/guides/freecodecamp/

**Comment by Luckerhns** [comment]
Офигенный простой гайд по NodeJS, спасибо огромное!

**Reply by Robinblad** [reply]
Благодарность автору (группе авторов) за сей титанический труд. Текст подлинно въедливо и кропотливо составлен. Здоровья вам и бодрости духа!

**Reply by bndby** [reply]
На здоровье! Возвращайтесь за добавкой ;-)

**Comment by shues** [comment]
Огромное спасибо, это большая и полезная работа. Некоторые есть места где видимо не совсем точно или возможно дословно переведено, но это реально крутое руководство для начинающих в node. Еще раз огромное спасибо.

**Comment by Bahurtsev** [comment]
Спасибо друзья! Благодаря всем этим статьям было лучше понять что там у себя в документации написали американцы. Здесь в последнем примере вот что заметил: 

app.configure(), не является стандартным для Express.js. Вместо этого мы должны использовать переменную окружения NODE_ENV для определения режима работы приложения.
Создаём файл .env в нём пишем NODE_ENV=production
После в index.js пишем следующее:

````
let dotenv = require('dotenv'); // Данный модуль найдёт файл .env и загрузит переменные в объект process.env, описанные в нём. После установки этого модуля с помощью мидделвар dotenv.config() мы можем вытаскивать наши переменные как: process.env.мояПеременная.
dotenv.config();	// Запускаем модуль dotenvn с помощью которого мы подгружаем переменные из файла .env в объект process.env.

// При запуске сервера с помощью pm2 start index.js и файла .env в котором установлена переменная NODE_ENV=production мы запустим режим продакшен.
if(process.env.NODE_ENV === 'production'){	// Запутится этот режим.
	app.use('/', (req, res, next) => {	// Именно для этого режима мы можем отдельно написать свои функции.
		console.log('Включен режим продакшен.');
		next();
	});
}
else if(process.env.NODE_ENV === 'development'){	// А для другого режима другие функции.
	app.use('/', (req, res, next) => {
		console.log('Включен режим разработки.');
		next();
	});
}

// А так же для всех режимов сразу.
app.get('/', (req, res) => {
	res.render('index');
});
````

**Comment by YuraKissil** [comment]
Всё отлично!!!
Но вот кусок кода в разделе "Основы работы с MySQL и Node.js" предлагаю поправить

```javascript
const todo = {
  thing: 'Buy the milk',
  author: 'Flavio',
}

connection.query('INSERT INTO todos SET ?', todo, (error, results, fields) => {
  if (error) {
    console.error('An error occurred while executing the query');
    throw error;
  }

  const id = results.resultId;
  console.log(id);
});
```
